control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -135,17 +142,31 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 
 	topicName := topic(TopicPrefix, channel)
 
-	saramaConfig, err := kafka.GetClusterAdminSaramaConfig(saramaSecurityOption)
+	kafkaClusterAdminSaramaConfig, err := kafka.GetSaramaConfig(saramaSecurityOption)
 	if err != nil {
 		return statusConditionManager.FailedToCreateTopic(topicName, fmt.Errorf("error getting cluster admin sarama config: %w", err))
 	}
 
-	kafkaClusterAdmin, err := r.NewKafkaClusterAdmin(topicConfig.BootstrapServers, saramaConfig)
+	// Manually commit the offsets in KafkaChannel controller.
+	// That's because we want to make sure we initialize the offsets within the controller
+	// before dispatcher actually starts consuming messages.
+	kafkaClientSaramaConfig, err := kafka.GetSaramaConfig(saramaSecurityOption, kafka.DisableOffsetAutoCommitConfigOption)
+	if err != nil {
+		return statusConditionManager.FailedToCreateTopic(topicName, fmt.Errorf("error getting cluster admin sarama config: %w", err))
+	}
+
+	kafkaClusterAdmin, err := r.NewKafkaClusterAdmin(topicConfig.BootstrapServers, kafkaClusterAdminSaramaConfig)
```

Comment:
Perhaps we should name this to `NewKafkaClusterAdminClient` ? I mean the `ClusterAdmin` in Sarama land is the administrative client for Kafka. 

It always confuses me when looking at code related to this (`NewKafkaClusterAdmin` VS `NewKafkaClient`).

I guess upstream saram is named poor there, but perhaps for us we could do the change, and make that `NewKafkaClusterAdminClient` 

I am happy to do that refactors

Reply:
Thanks for https://github.com/knative-sandbox/eventing-kafka-broker/pull/1398